darts/timeseries.py

```diff
@@ -1223,217 +1584,204 @@ def map(self,
                           logger)
 
         if num_args == 1:  # simple map function f(x)
-            new_dataframe = self.pd_dataframe().applymap(fn)
+            df = self.pd_dataframe().applymap(fn)
         elif num_args == 2:  # map function uses timestamp f(timestamp, x)
             def apply_fn_wrapper(row):
                 timestamp = row.name
                 return row.map(lambda x: fn(timestamp, x))
-
-            new_dataframe = self.pd_dataframe().apply(apply_fn_wrapper, axis=1)
+            df = self.pd_dataframe().apply(apply_fn_wrapper, axis=1)
         else:
+            df = None
             raise_log(ValueError("fn must have either one or two arguments"), logger)
 
-        return TimeSeries(new_dataframe, self.freq_str())
+        return TimeSeries.from_dataframe(df)
 
     def to_json(self) -> str:
         """
         Converts the `TimeSeries` object to a JSON String
 
+        At the moment this function works only on deterministic time series (i.e., made of 1 sample).
```

Comment:
If we only support it for deterministic, we might also support to_csv?

Reply:
Yes good idea, but a bit out of scope for this PR (I've only translated these methods from the old class).